src/tqec/compile/compile.py

```diff
@@ -0,0 +1,282 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+from typing import Literal
+import cirq
+import stim
+import stimcirq
+
+from tqec.circuit.operations.measurement import Measurement
+from tqec.circuit.operations.operation import make_observable
+from tqec.circuit.operations.transformer import transform_to_stimcirq_compatible
+from tqec.circuit.schedule import ScheduledCircuit
+from tqec.compile.observables import (
+    get_center_qubit_at_horizontal_pipe,
+    get_midline_qubits_for_cube,
+    get_stabilizer_region_qubits_for_pipe,
+)
+from tqec.compile.substitute import (
+    SubstitutionKey,
+    SubstitutionRule,
+    DEFAULT_SUBSTITUTION_RULES,
+)
+from tqec.exceptions import TQECException
+from tqec.noise_models.base import BaseNoiseModel
+from tqec.plaquette.plaquette import RepeatedPlaquettes
+from tqec.position import Direction3D, Position3D
+from tqec.sketchup.block_graph import AbstractObservable, BlockGraph, Cube
+from tqec.compile.block import CompiledBlock, TiledBlocks
+from tqec.compile.specs import DEFAULT_SPEC_RULES, CubeSpec, SpecRule
+from tqec.templates.scale import round_or_fail
+
+
+@dataclass
+class CompiledGraph:
+    """Represents a compiled block graph.
+
+    This class should be easy to scale and generate circuits directly.
+    """
+
+    block_graph: BlockGraph
```

Comment:
It seems weird to me that the `BlockGraph` instance is still part of the `CompiledGraph`. I think I would prefer to include the observables (computed at compilation) here and require that the earliest block starts at `z = 0`. That would remove the 2 needs for a `block_graph` instance. What do you think?

Reply:
Note that after the related commits, the quick start section in #322  needs a little change: move `observables` argument from `generate_stim_circuit` to `compile_block_graph`. If there is any problem, I'll fix it tomorrow.